### PR TITLE
Adds SRI

### PR DESCRIPTION
**Pull request by elenatanasoiu**
[Trello card](https://trello.com/c/oFhtO0Gm/146-enable-subresource-integrity-sri-on-government-frontend-l)

This will add security checks to verify that the content being served on GOV.UK hasn't been modified by third parties:

- adds sprockets-rails so we can use the SRI check feature
- adds `integrity: true` to javascript_include_tag and stylesheet_link_tag

Depends on:

- static: https://github.com/alphagov/static/pull/1008
- govuk_template: https://github.com/alphagov/govuk_template/pull/294

In order to use SRI, the browser expects CORS to be supported. This is why we need to add `crossorigin: true` alongside `integrity: true`. See more [here](https://hacks.mozilla.org/2015/09/subresource-integrity-in-firefox-43/)

### CODE REVIEW

**Comment by mcgoooo** [comment]
@elenatanasoiu can you fix up the tests on this please

File: Gemfile.lock [hunk 1]
```diff
@@ -297,4 +297,4 @@ RUBY VERSION
    ruby 2.3.1p112
 
 BUNDLED WITH
-   1.14.6
+   1.14.5
```
Comment (mcgoooo) [inline Gemfile.lock]: maybe upgrade your bundler?

File: app/presenters/contents_list.rb [hunk 2]
```diff
@@ -20,7 +20,8 @@ def contents_link(text, id)
         track_category: 'contentsClicked',
         track_action: 'leftColumnH2',
         track_label: id
-      })
+      }
+)
```
Comment (fofr) [inline app/presenters/contents_list.rb]: What's with the unrelated formatting changes in this PR?

File: Gemfile [hunk 3]
```diff
@@ -38,6 +39,7 @@ group :development, :test do
   gem 'govuk-lint'
   gem 'pry-byebug'
   gem 'jasmine-rails', '~> 0.14.0'
+  gem 'jasmine-core', '2.6.0'
```
Comment (fofr) [inline Gemfile]: What's this dependency for?

File: app/presenters/contents_list.rb (hunk 2 above)
Comment (elenatanasoiu) [inline app/presenters/contents_list.rb]: govuk-lint is complaining about this, so I've fixed it up here

File: Gemfile (hunk 3 above)
Comment (elenatanasoiu) [inline Gemfile]: Removed

File: app/presenters/contents_list.rb (hunk 2 above)
Comment (fofr) [inline app/presenters/contents_list.rb]: When running the linter in Jenkins it runs in [diff mode](https://github.com/alphagov/govuk-puppet/blob/7ded141088152004a54441ff36781ad7d3581737/modules/govuk_jenkins/files/var/lib/jenkins/groovy_scripts/govuk_jenkinslib.groovy#L388), meaning it should only check lines that have changed.

This change looks wrong:
```ruby
  def contents_link(text, id)
    link_to(
      text,
      "##{id}",
      data: {
        track_category: 'contentsClicked',
        track_action: 'leftColumnH2',
        track_label: id
      }
)
  end
```

Comment (fofr) [inline app/presenters/contents_list.rb]: Unrelated linting changes are usually put into separate commits/PRs.

**Comment by mcgoooo** [comment]
@fofr the linter is broken and picks up random changes now. The linter shoudln't really run in diff mode really.

**Comment by elenatanasoiu** [comment]
I'm testing this together with the other dependent commits on integration.

File: app/presenters/contents_list.rb (hunk 2 above)
Comment (h-lame) [inline app/presenters/contents_list.rb]: I agree with @fofr - most of the changes in this PR are unrelated, lets see if we can drop them and still keep the linter happy.

File: Gemfile [hunk 4]
```diff
@@ -24,6 +24,7 @@ gem 'rails-controller-testing', '~> 0.1'
 gem 'govuk_ab_testing', '~> 2.0'
 gem 'htmlentities', '4.3.4'
 gem 'statsd-ruby', '1.3.0', require: 'statsd'
+gem 'sprockets-rails'
```
Comment (h-lame) [inline Gemfile]: I think it's worth adding a version constraint like "> 3.2" here to make it clear that we get a version that supports the integrity attribute.  As it is, the app already uses sprockets-rails so this entry might be redundant without the version constraint.  I'd probably be happy with dropping this from the Gemfile too TBH.

Comment (elenatanasoiu) [inline Gemfile]: government-frontend is using sprockets-rails 3.1.1 at the moment, so I'll have to add an explicit version

Comment (elenatanasoiu) [inline Gemfile]: Although SRI was introduced in 3.1.0 so there's no need to bump it up to 3.2. I'll remove it instead.

Comment (h-lame) [inline Gemfile]: oh, my bad, I thought it was 3.2

**Comment by fofr** [comment]
What's the `crossorigin: "anonymous"` part for? Looked for details in PR or commit but couldn't see.

**Comment by elenatanasoiu** [comment]
In the description for [What is SRI](https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity) you can see it also adds `crossorigin="anonymous"` alongside the `integrity` attribute. 

In order to use SRI, the browser [expects CORS to be supported](https://hacks.mozilla.org/2015/09/subresource-integrity-in-firefox-43/):

>An important side note is that for Subresource Integrity to work, the CDN must support Cross-Origin Resource Sharing (CORS). The crossorigin attribute in the above code snippet enforces a CORS-enabled load. The anonymous value means that the browser should omit any cookies or authentication that the user may have associated with the domain. This prevents cross-origin data leaks, and also makes the request smaller.
